src/common/SearchService/SearchService.js

```diff
@@ -0,0 +1,48 @@
+// @ts-check
+import TrieSearch from 'trie-search';
+
+/**
+ * SearchService - Responsible for all the search operations
+*/
+class SearchService{
+    /**
+     * IDataRow - A single row of search data
+     * @typedef {Object} IDataRow
+     */
+     
+    /**
+     * IData - An Array of search data rows
+     * @typedef {IDataRow[]} IData
+     */   
+    
+    /**
+     * Instantiate a new SearchService
+     * @param {IData} data 
```

Comment:
Add a description for any parameter variables.
```suggestion
     * @param {IData} data - the entirety of the data to search
```